demo.sh

```diff
@@ -0,0 +1,215 @@
+#!/bin/sh
+
+set -eu
+
+# Constants
+ELASTIC_STACK_VERSION="9.1.3"
+ENV_OVERRIDE_FILE=".env.override"
+NAMESPACE="opentelemetry-operator-system"
+HELM_REPO_NAME="open-telemetry"
+HELM_REPO_URL='https://open-telemetry.github.io/opentelemetry-helm-charts'
+
+DEMO_RELEASE="my-otel-demo"
+DEMO_CHART="open-telemetry/opentelemetry-demo"
+
+KUBE_STACK_RELEASE="opentelemetry-kube-stack"
+KUBE_STACK_CHART="open-telemetry/opentelemetry-kube-stack"
+KUBE_STACK_VERSION='0.3.3'
+KUBE_STACK_VALUES_URL_CLOUD='https://raw.githubusercontent.com/elastic/elastic-agent/refs/tags/v'$ELASTIC_STACK_VERSION'/deploy/helm/edot-collector/kube-stack/values.yaml'
+KUBE_STACK_VALUES_URL_SERVERLESS='https://raw.githubusercontent.com/elastic/elastic-agent/refs/tags/v'$ELASTIC_STACK_VERSION'/deploy/helm/edot-collector/kube-stack/managed_otlp/values.yaml'
+SECRET_NAME='elastic-secret-otel'
+
+DOCKER_COLLECTOR_CONFIG_CLOUD='./src/otel-collector/otelcol-elastic-config.yaml'
+DOCKER_COLLECTOR_CONFIG_SERVERLESS='./src/otel-collector/otelcol-elastic-otlp-config.yaml'
+COLLECTOR_CONTRIB_IMAGE=docker.elastic.co/elastic-agent/elastic-agent:$ELASTIC_STACK_VERSION
+
+
+# Variables
+deployment_type=""
+platform=""
+destroy="false"
+elasticsearch_endpoint=""
+elasticsearch_api_key=""
+
+usage() {
+    echo "Usage: $0 [cloud-hosted|serverless] [docker|k8s] | destroy [docker|k8s]"
+    exit 1
+}
+
+parse_args() {
+  if [ $# -eq 0 ]; then
+    usage
+  fi
+
+  if [ "$1" = "destroy" ]; then
+    destroy="true"
+    if [ $# -ge 2 ]; then
+      platform="$2"
+    fi
+    return
+  fi
+
+  deployment_type="$1"
+  if [ $# -ge 2 ]; then
+    platform="$2"
+  fi
+}
+
+update_env_var() {
+  VAR="$1"
+  VAL="$2"
+  if grep -q "^$VAR=" "$ENV_OVERRIDE_FILE"; then
+    sed -i '' "s|^$VAR=.*|$VAR=\"$VAL\"|" "$ENV_OVERRIDE_FILE"
+  else
+    echo "$VAR=\"$VAL\"" >> "$ENV_OVERRIDE_FILE"
+  fi
+}
+
+ensure_env_values() {
+  if [ -z "$elasticsearch_endpoint" ]; then
+    printf "Enter your Elastic endpoint: "
```

Comment:
What do you think to include the mOTEL keyword incase of using serverless? (`Enter your Elastic OTLP endpoint`)

Reply:
That's a good point, I'll add it